src/toil_scripts/batch_alignment/bwa_alignment.py

```diff
@@ -1,223 +1,254 @@
-"""
-UC Santa Cruz Computational Genomics Lab
-Updated: 4/21/16
-Authors: John Vivian, Frank Nothaft
-
-Alignment of fastq reads via BWAkit
-
-    0 --> 1 -->(batching)--> 2 --> 3
-
-0   Download shared files
-1   Parse config
-2   Download sample
-3   Run BWAkit
-
-===================================================================
-:Dependencies:
-curl            - apt-get install curl
-Toil            - pip install toil
-Docker          - http://docs.docker.com/engine/installation/
-
-Optional:
-S3AM            - pip install --s3am (requires ~/.boto config file)
-"""
+#!/usr/bin/env python2.7
 import argparse
-import logging
 import multiprocessing
 import os
 import sys
 import textwrap
+from urlparse import urlparse
 
 import yaml
 from toil.job import Job
+
 from toil_scripts.lib import require
-from toil_scripts.lib.files import move_files
+from toil_scripts.lib.files import move_file_job
 from toil_scripts.lib.jobs import map_job
-from toil_scripts.lib.programs import docker_call
-from toil_scripts.lib.urls import download_url_job, s3am_upload
+from toil_scripts.lib.urls import download_url_job, s3am_upload_job
 from toil_scripts.rnaseq_cgl.rnaseq_cgl_pipeline import generate_file
+from toil_scripts.tools.aligners import run_bwakit
+from toil_scripts.tools.preprocessing import run_samtools_faidx, run_bwa_index
 
-_log = logging.getLogger(__name__)
 
-def download_shared_files(job, inputs, sample, output_dir, rg_line=None):
+def download_reference_files(job, inputs, samples):
     """
     Downloads shared files that are used by all samples for alignment
 
     :param JobFunctionWrappingJob job: passed automatically by Toil
-    :param JobFunctionWrappingJob job: Passed by Toil automatically
     :param Namespace inputs: Input arguments (see main)
+    :param list[list[str, list]] samples: Sample UUIDs and URLs
```

Comment:
Ah, I had it flipped. Disregard. Thanks for the clarification, @hannes-ucsc.

> If this is a doubly nested list containing string/list tuples, I think it needs more of a description.

Still holds, though.